ONSdigital/dp-dataset-api
# Remove auditing

### What

- Removed auditing (main task):
  - code, tests and dependencies
  - Kafka producer
  - Audit topic config
- Moved dependencies from go-ns to dp-net
  - Identity handlers from dp-net, depends on dp-net PR https://github.com/ONSdigital/dp-net/pull/12
  - Upgraded dependencies to versions that use dp-net ContextKey
- Introduced service package (with unit tests), to be consistent with newer APIs code structure
  - moved initialiser to service package
  - created more interfaces, so that service is testable
  - added session and health client to mongo struct in order to have a cleaner interface
  - moved most of main.go code to service.go (similar to other APIs code)
  - moved http server, midelware, etc from api.go to service.go
  - identity client created only once (before it was created once for healthcheck and once for the identity middleware)

### How to review

- Make sure code changes make sense
- Check that new code organisation makes sense (similar to other APIs)
- Unit tests should pass

### Who can review

Anyone
